components/match2/wikis/rainbow_six/get_match_group_copy_paste_wiki.lua

```diff
@@ -0,0 +1,121 @@
+}
+
+local VETOES = {
+	[0] = '',
+	[1] = 'ban,ban,ban,decider',
+	[2] = 'ban,ban,pick,ban',
+	[3] = 'ban,pick,ban,decider',
+	[4] = 'pick,ban,pick,ban',
+	[5] = 'pick,ban,pick,decider',
+	[6] = 'pick,pick,pick,ban',
+	[7] = 'pick,pick,pick,decider',
+	[8] = 'pick,pick,pick,pick,ban',
+	[9] = 'pick,pick,pick,pick,decider',
+}
+
+--default opponent type (used if the entered mode is not found in the above table)
+local DefaultMode = 'team'
+
+--returns the cleaned opponent type
+function wikiCopyPaste.getMode(mode)
+	return MODES[string.lower(mode or '')] or DefaultMode
+end
+
+--returns the Code for a Match, depending on the input
+function wikiCopyPaste.getMatchCode(bestof, mode, index, opponents, args)
+	local score = args.score == 'true' and '|score=' or ''
+	local mapDetails = args.detailedMap == 'true'
+	local mapDetailsOT = args.detailedMapOT == 'true'
+	local mapVeto = args.mapVeto == 'true'
+	local streams = args.streams == 'true'
+	local out = '{{Match' .. '\n\t|date=|finished='
+
+	if streams then
+		out = out .. '\n\t|twitch=|youtube=|vod='
+	end
+
+	for i = 1, opponents do
+		out = out .. '\n\t|opponent' .. i .. '=' .. wikiCopyPaste._getOpponent(mode, score)
+	end
+
+	if mapVeto and VETOES[bestof] then
+		out = out .. '\n\t|mapveto={{MapVeto'
+		out = out .. '\n\t\t|firstpick='
+		out = out .. '\n\t\t|types=' .. VETOES[bestof]
+		out = out .. '\n\t\t|t1map1=|t2map1='
+		out = out .. '\n\t\t|t1map2=|t2map2='
+		out = out .. '\n\t\t|t1map3=|t2map3='
+		out = out .. '\n\t\t|decider='
+		out = out .. '\n\t}}'
+	end
+	for i = 1, bestof do
+		out = out .. '\n\t|map' .. i .. '={{Map|map='
+		if not mapDetails then
+			out = out .. '|score1=|score2='
+		end
+		out = out .. '|finished='
+		if mapDetails then
+			out = out .. '\n\t\t|t1ban1=|t1ban2='
+			out = out .. '\n\t\t|t2ban1=|t2ban2='
+			out = out .. '\n\t\t|t1firstside='
+			if mapDetailsOT then
+				out = out .. '|t1firstsideot='
+			end
+			out = out .. '\n\t\t|t1atk=|t1def='
+			if mapDetailsOT then
+				out = out .. '|t1otatk=|t1otdef='
+			end
+			out = out .. '\n\t\t|t2atk=|t2def='
+			if mapDetailsOT then
+				out = out .. '|t2otatk=|t2otdef='
+			end
+			out = out .. '\n\t'
+		end
+		out = out .. '}}'
+	end
+
+	return out .. '\n\t}}'
+end
+
+--subfunction used to generate the code for the Opponent template, depending on the type of opponent
+function wikiCopyPaste._getOpponent(mode, score)
+	local out
+
+	if mode == '1v1' then
+		out = '{{PlayerOpponent|p1=' .. score .. '}}'
+	elseif mode == 'team' then
+		out = '{{TeamOpponent|' .. score .. '}}'
+	elseif mode == 'literal' then
+		out = '{{LiteralOpponent|}}'
+	end
+
+	return out
+end
+
+--function that sets the text that starts the invoke of the MatchGroup Moduiles,
+--contains madatory stuff like bracketid, templateid and MatchGroup type (matchlist or bracket)
+--on sc2 also used to link to the documentation pages about the new bracket/match system
+function wikiCopyPaste.getStart(template, id, modus, args)
```

Comment:
that function seems to have sc2 specific links in it, or does R6 plan to create/update the pages that get linked here?
additionally the singlematch option is missing